### PR TITLE
Support SRM NuGet packages

### PR DESCRIPTION
**Pull request by MichielOda**

### CODE REVIEW

File: Assemblers.Common/PackageReferenceProcessor.cs [hunk 1]
```diff
@@ -184,13 +179,14 @@ private static bool IsDevPackNuGetPackage(string packageId)
...
-        private static IEnumerable<SourcePackageDependencyInfo> GetResolvedPackages(ISourceRepositoryProvider sourceRepositoryProvider,
+        private static IEnumerable<PackageIdentity> GetResolvedPackages(ISourceRepositoryProvider sourceRepositoryProvider,
                                                                               ILogger logger, IList<PackageIdentity> extensions,
                                                                               HashSet<SourcePackageDependencyInfo> allPackagesAndDependencies)
         {
             if (allPackagesAndDependencies.Count == 0)
             {
-                return Array.Empty<SourcePackageDependencyInfo>();
+                // If no dependencies, then return the main package(s) at least.
+                return extensions;
```
Comment (janstaelensskyline) [inline Assemblers.Common/PackageReferenceProcessor.cs:189]: Not sure I understand this. Can you explain the logic?

File: Assemblers.Common/PackageReferenceProcessor.cs [hunk 2]
```diff
@@ -300,11 +304,13 @@ private void ProcessResolvedPackages(IEnumerable<SourcePackageDependencyInfo> re
                             // Needs to be added as a reference in the dllImport attribute/script references.
                             nugetPackageAssemblies.DllImportNugetAssemblyReferences.Add(packageAssemblyReference);
 
-                            if (!resolvedPackage.Id.StartsWith(DevPackHelper.FilesPrefix))
+                            if (dontAddToPackageToInstall)
```
Comment (janstaelensskyline) [inline Assemblers.Common/PackageReferenceProcessor.cs:307]: I might be reading wrong, but this will always be false.    
dontAddToPackageToInstall  = false;
if ()
  set to true
else if()
  set to true
else()
 check condition
 
 if it reaches the else, it never execute the if or else if code. So it will always be false.

**Review by janstaelensskyline — COMMENTED** [review]
Few questions. Also need to still take a look at following the complete flow in the solution. Hard to follow with just the changed files.

File: Assemblers.Common/PackageReferenceProcessor.cs (hunk 1 above)
Comment (MichielOda) [reply Assemblers.Common/PackageReferenceProcessor.cs:189]: This method returns the list of all the NuGet packages that are needed. If there are no dependencies, it would return an empty list, resulting in nothing. As for SRM we ignore all the dependencies, if you only have that NuGet, then it would return nothing. So the further logic wouldn't know about it and would be unable to add the script reference or dllImport.

**Review by janstaelensskyline — APPROVED** [review]
